packages/c/sshnpd/src/main.c

```diff
@@ -100,9 +50,30 @@ static void child_exit_handler(int sig) {
   }
 }
 
+static void free_if_not_null(void *ptr) {
+  if (ptr != NULL) {
+    free(ptr);
+    ptr = NULL;
+  }
+}
+
 int main(int argc, char **argv) {
   int res = 0;
-  int exit_res = 0;
+
+  // setup initial values for global variables
+  is_child_process = false;
+  should_run = 1;
+  device_info_pos = 0;
+  device_info_last_sent = NULL;
+  device_info_attempts = 0;
+
+  // device info
+  struct atcommons_memlist memlist = atcommons_memlist_create(32);
+#define safe_memlist(x) res = x;
+  if (res > 0) {
+    atcommons_memlist_failure_free(&memlist);
+    return res;
+  }
```

Comment:
Now using atcommons_memlist to manage all of the setup & teardown of memory in the main process.